### PR TITLE
CMake: fix cuda detection and jenkins setup 

### PR DESCRIPTION
**Pull request by anstaf**

### CODE REVIEW

**Comment by anstaf** [comment]
launch strgrid

**Comment by anstaf** [comment]
launch strgrid

**Comment by anstaf** [comment]
launch strgrid

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
launch strgrid

**Comment by havogt** [comment]
latest changes broke clang cuda detection

**Comment by havogt** [comment]
Problem is:
- with clang 9, nvcc is not compatible and cannot be used to search for the toolkit
- find_package(CUDA) cannot be used as it conflicts with enable_language

We should see if we can steal https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#module:FindCUDAToolkit from cmake 3.17 or just bump the required version to 3.17 once it is released.

**Comment by havogt** [comment]
launch strgrid

File: CMakeLists.txt [hunk 1]
```diff
@@ -126,7 +116,7 @@ if (TARGET gridtools_cuda)
     target_link_libraries(storage_cuda INTERFACE gridtools gridtools_cuda)
     list(APPEND GT_STORAGES cuda)
 
-    if(TARGET MPI::MPI_CXX)
+    if(MPI_CXX_FOUND)
```
Comment (anstaf) [inline CMakeLists.txt]: Is there the difference between `if(MPI_CXX_FOUND)` and  `if(TARGET MPI::MPI_CXX)` ? If not I would prefer the `TARGET` form to minimize the set of used global variables.

File: jenkins/envs/daint_clang.sh [hunk 2]
```diff
@@ -5,8 +5,9 @@ source $(dirname "$BASH_SOURCE")/daint.sh
 module load /users/vogtha/modules/compilers/clang/7.0.1
 module load gcc
 
+export GTCMAKE_GT_PREFER_CLANG_CUDA_OVER_NVCC_CUDA=OFF
```
Comment (anstaf) [inline jenkins/envs/daint_clang.sh]: why?

File: CMakeLists.txt (hunk 1 above)
Comment (havogt) [reply CMakeLists.txt]: This is probably a bug in the FindMPI: the target is set even if detection fails.

File: jenkins/envs/daint_clang.sh (hunk 2 above)
Comment (havogt) [reply jenkins/envs/daint_clang.sh]: The clang version which is used here (in my home directory) doesn't work with clang cuda. Also previously this plan was for testing nvcc with clang as host compiler, so now it tests the same thing as before.

**Pull request merged**